### PR TITLE
Fixes issue #404.

### PR DESCRIPTION
**Pull request by supercooldave**
There was a problem compiling in the encore standard library directory -- the compiler would look for modules in the standard directory and in ./ and register an ambiguity. 

This fix checks to see whether ./ is the same as the standard module directory and if so removes it from the directories scanned.

Testing involves going into the standard library directory and compiling `String.enc` with the `-tc` flag and seeing that the previous error is no longer there.

Addresses issue #404.

### CODE REVIEW

File: src/types/Types.hs [hunk 1]
```diff
@@ -187,7 +187,7 @@ modeSubtypeOf ty1 ty2 =
   where
     hasMatchingMode modes mode
       | modeIsSharable mode = mode `elem` modes || Sharable `elem` modes
-      | otherwise = mode `elem` modes || Unsafe `elem` modes
+      | otherwise = mode `elem` modes
```
Comment (EliasC) [inline src/types/Types.hs]: Why was this check removed?

File: src/tests/encore/parametric/unsafe.enc [hunk 2]
```diff
@@ -11,9 +10,9 @@ end
 active class Main
   def main(args : [String]) : unit
     val duplicator = new Duplicator[Token]
-    duplicator.f = new Token
-    match duplicator.duplicate() with
-      case (y, y') => println("All your linear are belong to us")
+    var x = new Token
+    match duplicator.duplicate(x) with
+      case (y, y') => println("all your linear are belong to us")
```
Comment (EliasC) [inline src/tests/encore/parametric/unsafe.enc]: This commit seems to change things that it shouldn't change...

File: src/front/ModuleExpander.hs [hunk 3]
```diff
@@ -121,9 +121,13 @@ buildModulePath (NSExplicit ns) =
 findSource :: [FilePath] -> FilePath -> ImportDecl -> IO FilePath
 findSource importDirs sourceDir Import{itarget} = do
   let modulePath = buildModulePath itarget
-      sources = nub $
-                sourceDir </> modulePath :
-                map (</> modulePath) importDirs
+      imports = map (</> modulePath) importDirs
+      sourceModulePath = sourceDir </> modulePath 
+  expandedSourceModulePath <- makeAbsolute $ sourceModulePath
+  let sources = if expandedSourceModulePath `elem` imports then -- in std library directory
```
Comment (EliasC) [inline src/front/ModuleExpander.hs]: What in this line ensures that we are looking at imports of the standard library?

**Review by EliasC — CHANGES_REQUESTED** [review]
This commit seems to revert the changes of previous commits. I suspect there's been a merge issue.

File: src/types/Types.hs (hunk 1 above)
Comment (supercooldave) [reply src/types/Types.hs]: Oo. I probably forgot to rebase. 
I'll fix. Same with point below.

File: src/front/ModuleExpander.hs (hunk 3 above)
Comment (supercooldave) [reply src/front/ModuleExpander.hs]: Technically, it checks that the source directory (of the compilation target) is not among the import directories (including the std libs), and if it is it removes it. 
This is a more robust solution, IMO, but I should document the code to reflect this fact.

**Comment by supercooldave** [comment]
@EliasC I've recommitted. It seems that I'd done a revert to undo some of my changes, but it undid your changed. I checked out afresh and re-added this file. The version now has a comment that hopefully explains the logic.

**Review by EliasC — APPROVED** [review]

**Comment by EliasC** [comment]
This looks good! Merging tomorrow.

**Pull request merged**